File: backend/app/models/allarme.py
from typing import Literal

from pydantic import BaseModel

Livello = Literal["Minor", "Warning", "Major"]


class AllarmeOut(BaseModel):
    id: int
    stato: bool
    impianto: str
    azienda: str
    gruppo: str
    dispositivo: str
    livello: Livello
    dataInizio: str
    dataFine: str | None
    durata: str | None
    descrizione: str

File: frontend/src/types/allarme.ts
export type Livello = 'Minor' | 'Warning' | 'Major';

export interface Allarme {
    id: number;
    stato: boolean;
    impianto: string;
    azienda: string;
    gruppo: string;
    dispositivo: string;
    livello: Livello;
    dataInizio: string;
    dataFine?: string | null;
    durata?: string | null;
    descrizione: string;
}